src/resmom/linux/alps.c

```diff
@@ -5817,15 +5817,61 @@ alps_cancel_reservation(job *pjob)
 {
 	char buf[1024];
 	basil_response_t *brp;
+	char	filename[MAXPATHLEN];
 
 	if (!pjob) {
 		log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_NODE, LOG_NOTICE, __func__,
 			"Cannot cancel ALPS reservation, invalid job.");
 		return (-1);
 	}
+
+	/*
+	 * Get the name of the file that contains ALPS reservation ID
+	 * information.  We may need to read the file, or we just need to
+	 * unlink it.  Either way, we need the filename.
+	 */
+	snprintf(filename, MAXPATHLEN, "%s/aux/%s.alpsresv", pbs_conf.pbs_home_path, pjob->ji_qs.ji_jobid);
```

Comment:
Shouldn't this snprintf be within the if below?

Reply:
Even if we don't read the file, we need the filename to delete the file. That's why it's outside the if.